### DISCUSSION TITLE
Identifying the direction of change

### DISCUSSION THREAD

**Discussion by Meligui** [deepcharles/ruptures#281] [Q&A] [answered]
Hi, I was using the library for an application where I would also like to identify the direction of the change (i.e., at the change point whether the algorithm is detecting increased or decreased values of the time series in question)

I thought of implementing it by just checking averages of a few samples before and after the change point and identifying the change direction this way but was wondering if there is a better way to do it using the library itself. 

I looked through the documentation but couldn't find a way.

Thanks.

**Comment by deepcharles** [comment, marked as answer]
Hi, thanks for your interest in ruptures.

It is not coded in the library by it can easily be done:

```python
import ruptures as rpt

def triplewise(iterable):
    "Return overlapping triplets from an iterable"
    # triplewise('ABCDEFG') --> ABC BCD CDE DEF EFG
    for (a, _), (b, c) in rpt.pairwise(rpt.pairwise(iterable)):
        yield a, b, c

# assume a variable `signal` and a list of change-points `bkps`
for (left, mid, right) in triplewise([0] + bkps):
    left_mean = signal[left:mid].mean()
    right_mean = signal[mid:right].mean()
    if left_mean > right_mean:
        print(f"Increase at {mid}")
    else:
        print(f"Decrease at {mid}")

```

**Reply by Meligui** [reply]
Thanks a lot for your answer, also thanks a lot for the library it is really useful great work.

One other question, what does pairwise() do? can't seem to find it in the documentation.

**Reply by deepcharles** [reply]
It is a small util function that I often use: it takes an iterable and return the successive overlaping pairs, i.e. `pairwise(["A", "B", "C", "D"]) --> ("A", "B") ("B", "C")  ("C", "D")`.
Super useful when you want to iterate over segments when you have a list of change-points.

**Reply by Meligui** [reply]
Thanks again for the quick response, will try your suggested solution out.
